autobahn/asyncio/websocket.py

```diff
@@ -194,18 +194,21 @@ class WebSocketServerProtocol(WebSocketAdapterProtocol, protocol.WebSocketServer
     def _onConnect(self, request):
         # onConnect() will return the selected subprotocol or None
         # or a pair (protocol, headers) or raise an HttpException
-        ##
-        # noinspection PyBroadException
-        try:
-            res = self.onConnect(request)
-            if yields(res):
-                asyncio.async(res)
-        except ConnectionDeny as e:
-            self.failHandshake(e.reason, e.code)
-        except Exception as e:
-            self.failHandshake("Internal server error: {}".format(e), ConnectionDeny.INTERNAL_SERVER_ERROR)
+        fut = txaio.as_future(self.onConnect, request)
+        fut.add_done_callback(self._on_connect_callback)
+
```

Comment:
Basically all operations on Futures/Deferreds should use txaio. So this should probably be`txaio.add_callbacks(fut, self._on_connect_callback, self._on_connect_error)` or similar. However, see discussion in the bugs (and, it's also "okay-ish" to use asyncio-only APIs in this particular module because it's `autobahn.asyncio.*` and therefore shouldn't even be imported on non-asyncio platforms -- i.e. when using Twisted).